crates/feldera-storage/src/backend/mod.rs

```diff
@@ -26,6 +26,28 @@ pub mod monoio_impl;
 #[cfg(test)]
 pub(crate) mod tests;
 
+/// An Increment Only Atomic.
+///
+/// Usable as a global counter to get unique identifiers for file-handles.
+///
+/// Because the buffer cache can be shared among multiple threads.
+#[derive(Debug, Default)]
+pub struct AtomicIncrementOnlyI64 {
+    value: AtomicI64,
+}
+
+impl AtomicIncrementOnlyI64 {
+    pub const fn new() -> Self {
+        Self {
+            value: AtomicI64::new(0),
+        }
+    }
+
+    pub(self) fn increment(&self) -> i64 {
```

Comment:
clippy doesn't like `pub(self)` because it is the same as not using `pub` at all, see https://doc.rust-lang.org/reference/visibility-and-privacy.html#pubin-path-pubcrate-pubsuper-and-pubself

Reply:
fixed